hdfs/src/main/scala/akka/stream/alpakka/hdfs/impl/writer/HdfsWriter.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.hdfs.impl.writer
+
+import akka.stream.alpakka.hdfs.FilePathGenerator
+import akka.stream.alpakka.hdfs.impl.writer.HdfsWriter._
+import akka.util.ByteString
+import org.apache.hadoop.fs.{FileSystem, Path}
+
+/**
+ * Internal API
+ */
+private[hdfs] trait HdfsWriter[W, I] {
+
+  protected lazy val output: W = create(fs, temp)
+
+  protected lazy val temp: Path = tempFromTarget(pathGenerator, target)
+
+  def moveToTarget(): Boolean = {
+    if (!fs.exists(target.getParent))
+      fs.mkdirs(target.getParent)
+    fs.rename(temp, target)
+  }
+
+  def sync(): Unit
+
+  def targetFileName: String = target.getName
+
+  def write(input: I, addNewLine: Boolean): Long
+
+  def rotate(rotationCount: Long): HdfsWriter[W, I]
+
+  protected def target: Path
+
+  protected def fs: FileSystem
+
+  protected def pathGenerator: FilePathGenerator
+
+  protected def create(fs: FileSystem, file: Path): W
+
+}
+
+private[writer] object HdfsWriter {
+
+  val NewLineByteArray: Array[Byte] = ByteString(System.getProperty("line.separator")).toArray
```

Comment:
Is it useful to use the system's separator? It could be provided via the settings instead.